integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -1570,6 +1592,139 @@ describe.skipIf(environment.missingServer, "Flexible sync", function () {
+          expect(this.realm.subscriptions.state).to.equal(SubscriptionsState.Pending);
+
+          expect(this.realm.subscriptions).to.have.length(1);
+        });
+
+        it("waits for objects to sync when timeout is longer", async function (this: RealmContext) {
+          expect(this.realm.subscriptions).to.have.length(0);
+
+          const peopleOver10 = this.realm.objects(FlexiblePersonSchema.name).filtered("age > 10");
+          await peopleOver10.subscribe({ behavior: WaitForSync.Always, timeout: this.timeout() });
+          expect(this.realm.subscriptions.state).to.equal(SubscriptionsState.Complete);
+
+          expect(this.realm.subscriptions).to.have.length(1);
+        });
+
+        it("does not wait for objects to sync when timeout is shorter", async function (this: RealmContext) {
+          expect(this.realm.subscriptions).to.have.length(0);
+
+          const peopleOver10 = this.realm.objects(FlexiblePersonSchema.name).filtered("age > 10");
+          await peopleOver10.subscribe({ behavior: WaitForSync.Always, timeout: 0 });
+          expect(this.realm.subscriptions.state).to.equal(SubscriptionsState.Pending);
+
+          expect(this.realm.subscriptions).to.have.length(1);
+        });
+
+        it("returns the same 'Results' instance", async function (this: RealmContext) {
+          const beforeSubscribe = this.realm.objects(FlexiblePersonSchema.name).filtered("age > 10");
+          const afterSubscribe = await beforeSubscribe.subscribe();
+          expect(beforeSubscribe).to.equal(afterSubscribe);
+        });
+      });
+
+      describe("Results#unsubscribe", function () {
+        it("unsubscribes from existing subscription", async function (this: RealmContext) {
+          const peopleOver10 = await this.realm.objects(FlexiblePersonSchema.name).filtered("age > 10").subscribe();
+          expect(this.realm.subscriptions).to.have.length(1);
+
+          const unsubscribed = peopleOver10.unsubscribe();
+          expect(unsubscribed).to.be.true;
+          expect(this.realm.subscriptions).to.have.length(0);
+        });
+
+        it("does not unsubscribe when there is no subscription", function (this: RealmContext) {
+          const peopleOver10 = this.realm.objects(FlexiblePersonSchema.name).filtered("age > 10");
+          expect(this.realm.subscriptions).to.have.length(0);
+
+          const unsubscribed = peopleOver10.unsubscribe();
+          expect(unsubscribed).to.be.false;
+          expect(this.realm.subscriptions).to.have.length(0);
+        });
+
+        it("does not unsubscribe multiple times", async function (this: RealmContext) {
+          const peopleOver10 = await this.realm.objects(FlexiblePersonSchema.name).filtered("age > 10").subscribe();
+          expect(this.realm.subscriptions).to.have.length(1);
+
+          let unsubscribed = peopleOver10.unsubscribe();
+          expect(unsubscribed).to.be.true;
+
+          unsubscribed = peopleOver10.unsubscribe();
```

Comment:
I agree that it may not be as clear a use case for what a user would do with the return value. But just to add some specificity to what you said, it returns whether it was unsubscribed which could return `false`, e.g. if the query was not subscribed to beforehand.

The Kotlin SDK is facing some issues implementing this and there are discussions if this particular API should just be removed for now. Users would then use the currently available API via `MutableSubscriptionSet`.

Reply:
Update: It now returns `void`.